certbot/plugins/manual.py

```diff
@@ -183,15 +211,30 @@ def _perform_single(self, achall):
 
         return response
 
-    def _notify_and_wait(self, message):  # pylint: disable=no-self-use
-        # TODO: IDisplay wraps messages, breaking the command
-        #answer = zope.component.getUtility(interfaces.IDisplay).notification(
-        #    message=message, height=25, pause=True)
-        sys.stdout.write(message)
-        six.moves.input("Press ENTER to continue")
+    def _perform_dns01_challenge(self, achall):
+        response, validation = achall.response_and_validation()
+        if not self.conf("test-mode"):
+            if not self.conf("public-ip-logging-ok"):
```

Comment:
This same set of logic is in `_perform_http01_challenge`, it should be factored out in to its own utility method, something like `_ip_logging_permission`.